### PR TITLE
diamond: init at 0.8.36

### PR DESCRIPTION
**Pull request by coissac**
###### Motivation for this change

This is a new package, DIAMOND, a bioinformatics tool to perform that implements an accelerated BLAST compatible local sequence aligner.

B. Buchfink, Xie C., D. Huson, Fast and sensitive protein alignment using DIAMOND, Nature Methods 12, 59-60 (2015).

###### Things done

- [ ] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by matthiasbeyer** [comment]
Any comment (in the package source) why the patch is necessary would be nice. Also: If this patch is needed to be able to compile the package, consider contributing this patch upstream!

**Comment by coissac** [comment]
The patch is just to remove two warnings during the compilation. I can send it to the authors but in the meantime where and how should I explain the patch ? in the default.nix file ?

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: Indent 2 spaces left.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: `gcc` should not be necessary. `cmake` should be in `nativeBuildInputs`.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: Indent whole `longDescription` two spaces right.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: Indent this paragraph 6 spaces right.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: Indent 4 space left.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: Indent 2 spaces right.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: Please remove the empty lines here.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: Are you sure this is necessary? This should be detected automatically.

File: pkgs/applications/science/biology/diamond/default.nix
Comment (fpletz) [inline pkgs/applications/science/biology/diamond/default.nix]: You could give the patch a more telling name like `fix-warnings.patch`.

**Review by fpletz — CHANGES_REQUESTED** [review]

**Comment by coissac** [comment]
Thank you for your comments. With the commit 20fd13d86caa4be6b872283d7038a3c2f7df05f3, I hope that I have fulfill all the requests from the previous review.

**Review by matthiasbeyer — CHANGES_REQUESTED** [review]
Maintainer, please do not merge. Reason: Look at commit history. It should be cleaned up first!

**Comment by coissac** [comment]
I had some problems because of a synchronization between the  NixOS:master and of metabarcoding:master. I tried to isolate the diamond package in a single commit by merging the different branches on our side before recreating the file and commit them in a single commit  : 20fd13d86caa4be6b872283d7038a3c2f7df05f3. Now I don't know how to "It should be cleaned up first!"

**Comment by matthiasbeyer** [comment]
> metabarcoding:master

what is that for a repository and why do you need it? And why do you need to stay up to date with nixos master branch for a new package?

**Comment by coissac** [comment]
Ok, it did a mistake, perhaps I didn't say it enough clearly. But now that the mistake exists how can I clean things to be in the right way ?

Thank you for your help.

**Comment by matthiasbeyer** [comment]
So it seems that your branch is based on 9ae8ce813e7ed78c138e432c2f04995d5f56dd62. I guess you can `git rebase -i 9ae8ce813e7ed78c138e432c2f04995d5f56dd62` your branch, then remove all the merge commits and other stuff and only leave commits where you added or removed content. From what I can see, this should be only 20fd13d86c.

Then, you can push this branch again...

**Comment by matthiasbeyer** [comment]
@fpletz I re-submitted this PR in https://github.com/NixOS/nixpkgs/pull/24840 - feel free to cherry-pick onto master. See initial PR message for what I did there.

**Comment by Mic92** [comment]
was merged in #24840

**Pull request closed without merging**
